libdnf/conf/ConfigMain.cpp

```diff
@@ -214,6 +214,7 @@ class ConfigMain::Impl {
     OptionBool obsoletes{true};
     OptionBool showdupesfromrepos{false};
     OptionBool exit_on_lock{false};
+    OptionBool allow_vendor_change{false};
```

Comment:
Shouldn't this default to `true`?

Reply:
This should be false in code, but the dnf.conf shipped in RHEL should set it to `true`.

Reply:
On the other hand... It might actually be quite desirable to change the behavior in RHEL and have this feature turned on there...

Reply:
It looks like that a vendor change is respected in libsolv for only upgrade operation. Therefore it doesn't sound like we can allow allow_vendor_change to be setted as `false` in default, because distrosync and downgrade command will work differently than upgrade command.
Also I don't see any benefit with allow_vendor_change as `false`. I see it as an additional complication for present users/customers. Please could anyone provide a big picture why we should add an additional restriction in default behaviour?